### PR TITLE
Fix config regression

### LINKED ISSUE #380: Regressions raising after bumping elemental-cli in elemental-toolkit
Now elemental-cli returns error in various points while loading the configuration files. Previously several of these errors were swallowed causing a fallback to defaults. Now errors are caught and it causes failures were it shouldn't. The obvious case is with `elemental install` when no configuration file is present, now it fails. This can be seen at https://github.com/rancher/elemental-toolkit/actions/runs/3498251946/jobs/5858917572

### PR DESCRIPTION
**Pull request by frelon**
Checks if config-file exists before trying to merge it into running configuration.

Fixes #380 

Signed-off-by: Fredrik Lönnegren <fredrik.lonnegren@suse.com>

### CODE REVIEW

File: cmd/config/config.go [hunk 1]
```diff
@@ -195,6 +196,7 @@ func ReadConfigRun(configDir string, flags *pflag.FlagSet, mounter mount.Interfa
 	err := viper.Unmarshal(cfg, setDecoder, decodeHook)
 	if err != nil {
 		cfg.Logger.Warnf("error unmarshalling RunConfig: %s", err)
+		return cfg, err
```
Comment (davidcassany) [inline cmd/config/config.go:199]: In fact I prefer to keep it. I liked the fact of being explicit on errors, we just need to handle them better.

The changes you did on tracking proper error messages uncovered poor previous logic

**Review by davidcassany — COMMENTED** [review]
I have been also working on fix 😅 I was about to submit it. In fact I am fighting with another corner case I saw. Unit tests for this part are a pretty annoying to code and I realized there are some glitches.

**Comment by frelon** [comment]
Duplicate of #382
